metomi/rose/date_cli.py

```diff
@@ -177,6 +177,40 @@
 
 from metomi.isodatetime.main import main as iso_main
 
+from metomi.rose.date import LEGACY_OFFSET, upgrade_offset
+
+
+def _handle_old_offsets(args: list) -> list:
+    """Handle Legacy Rose date --offset values:
+
+    # https://github.com/metomi/rose/issues/2577
+
+
+    Examples:
+    >>> _handle_old_offsets(['rose-date', '--offset=1d1s'])
+    [WARN] This offset syntax is deprecated.
+    ['rose-date', '--offset=P1DT0H0M1S']
+    >>> _handle_old_offsets(['rose-date', '-s', '1d1s'])
+    [WARN] This offset syntax is deprecated.
+    ['rose-date', '-s', 'P1DT0H0M1S']
+    """
+    for index in range(len(args)):
+        arg = args[index]
+        if '--offset' in arg or '-s' in arg:
+            # Case: --offset=<offset> is a single item in args list:
+            if (
+                len(arg.split('=')) > 1
+                and LEGACY_OFFSET.match(arg.split('=')[1])
+            ):
+                offset = upgrade_offset(arg.split("=")[1])
+                args[index] = f'{arg.split("=")[0]}={offset}'
+            # Case: --offset <offset> is two items in args list:
+            elif (
+                LEGACY_OFFSET.match(args[index + 1])
+            ):
+                args[index + 1] = upgrade_offset(args[index + 1])
```

Comment:
I found a couple of facetious inputs that could cause traceback:

```console
$ rose date 20000101 -s= -1d
WARNING: "rose date" is deprecated, use the "isodatetime" command.
[WARN] This offset syntax is deprecated.
Invalid ISO 8601 date representation: \-P1D
$ rose date 20000101 -s= 
Traceback (most recent call last):
IndexError: list index out of range
```

These aren't a big concern but it would be good to handle them if possible. I managed to get around them by rejigging the if statements a little (needs review/testing):

```suggestion
            if '=' in arg:
                if LEGACY_OFFSET.match(arg.split('=')[1]):
                    offset = upgrade_offset(arg.split("=")[1])
                    args[index] = f'{arg.split("=")[0]}={offset}'

            # Case: --offset <offset> is two items in args list:
            elif (
                len(args) > index + 1
                and LEGACY_OFFSET.match(args[index + 1])
            ):
                args[index + 1] = upgrade_offset(args[index + 1])
```